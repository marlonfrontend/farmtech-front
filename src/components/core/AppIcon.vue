<script setup lang="ts">
const props = defineProps({
  icon: String,
  size: {
    type: Number,
    default: 24
  }
})

defineExpose(props)
</script>

<template>
  <svg viewBox="0 0 24 24" :width="`${size}px`" :height="`${size}px`">
    <use
      xmlns:xlink="http://www.w3.org/1999/xlink"
      :xlink:href="`/icons.svg#${icon}`"
      :href="`/icons.svg#${icon}`"
    />
  </svg>
</template>
