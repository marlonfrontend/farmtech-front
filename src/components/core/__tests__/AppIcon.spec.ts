import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import AppIcon from '../AppIcon.vue'

describe('AppIcon.vue', () => {
  it('should render component props', () => {
    const props = {
      icon: 'search'
    }
    const wrapper = mount(AppIcon, {
      propsData: { ...props }
    })
    expect(wrapper.props()).toContain(props)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
