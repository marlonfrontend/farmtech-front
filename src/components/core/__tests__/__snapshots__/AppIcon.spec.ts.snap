// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppIcon.vue > should render component props 1`] = `
"<svg viewBox=\\"0 0 24 24\\" width=\\"24px\\" height=\\"24px\\">
  <use xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" xlink:href=\\"/icons.svg#search\\" href=\\"/icons.svg#search\\"></use>
</svg>"
`;
